lib/sycamore/sycamore/writers/base.py

```diff
@@ -0,0 +1,160 @@
+from dataclasses import dataclass
+from typing import Any, Iterable, Optional
+
+from abc import ABC, abstractmethod
+import numpy as np
+
+from ray.data import Dataset, Datasink
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.execution.interfaces import TaskContext
+from ray.data.block import Block, BlockAccessor
+from sycamore.data.document import Document, MetadataDocument
+from sycamore.plan_nodes import Node, Write
+
+
+class BaseDBWriter(Write):
+
+    # Type param for the client
+    class client_t(ABC):
```

Comment:
Based on my understanding of python naming, I'd think this should be ClientT, but I haven't seen people using the T convention in python, so I'd go with either just Client, or AbstractClient.

(same below)